### PR TITLE
Dynaconf conversion for docker settings

### PR DESCRIPTION
**Pull request by tpapaioa**
This PR converts the `docker` section of `robottelo.properties` to `dynaconf` settings.

The `docker_image` setting has been removed, and new repo settings `settings.repos.rhel7_optional` and `settings.repos.rhel7_extras` have been added. With the recent pytest conversion, the `docker_host` fixture now installs the `docker` rpm on a rhel7 vm deployed with `broker`, instead of deploying the RHV snapshot previously stored in `docker_image`.

### CODE REVIEW

File: tests/foreman/cli/test_docker.py [hunk 1]
```diff
@@ -37,50 +37,101 @@
...
-from robottelo.test import CLITestCase
-from robottelo.vm import VirtualMachine
 
-DOCKER_PROVIDER = 'Docker'
-REPO_CONTENT_TYPE = 'docker'
-REPO_UPSTREAM_NAME = 'busybox'
+BASE_URL = 'http://download.eng.rdu2.redhat.com/rhel-7/rel-eng'
+SERVER_REPO = f'{BASE_URL}/updates/RHEL-7/latest-RHEL-7.9/compose/Server/x86_64/os/'
+OPTIONAL_REPO = f'{BASE_URL}/RHEL-7/latest-RHEL-7/compose/Server-optional/x86_64/os/'
+EXTRAS_REPO = f'{BASE_URL}/EXTRAS-7/latest-EXTRAS-7.9-RHEL-7/compose/Server/x86_64/os/'
```
Comment (jyejare) [inline tests/foreman/cli/test_docker.py:51]: Immediately remove them from here as we are exposing RedHats internal info here. Ask these repos from repos.yaml OR robottelo section in robottelo.properties !

File: tests/foreman/cli/test_docker.py [hunk 2]
```diff
@@ -37,50 +37,101 @@
...
         }
     )
 
 
-class DockerManifestTestCase(CLITestCase):
+def _content_view(repo_id, org_id):
+    """Create a content view and link it to the given repository."""
+    content_view = make_content_view({'composite': False, 'organization-id': org_id})
+    ContentView.add_repository({'id': content_view['id'], 'repository-id': repo_id})
+    return ContentView.info({'id': content_view['id']})
```
Comment (jyejare) [inline tests/foreman/cli/test_docker.py:83]: For test setup purpose you can use Nailgun methods as they are faster that CLI.

File: tests/foreman/cli/test_docker.py [hunk 3]
```diff
@@ -37,50 +37,101 @@
...
+
+@pytest.fixture
+def content_view_promote(content_view_publish, module_lce):
+    ContentView.version_promote(
+        {
+            'id': content_view_publish['id'],
+            'to-lifecycle-environment-id': module_lce.id,
+        }
+    )
+    return ContentView.version_info({'id': content_view_publish['id']})
```
Comment (jyejare) [inline tests/foreman/cli/test_docker.py:124]: For test setup purpose you can use nailgun existing fixtures as they are faster than CLI.

File: tests/foreman/cli/test_docker.py [hunk 4]
```diff
@@ -37,50 +37,101 @@
...
+
+@pytest.fixture
+def repo(module_product):
+    return _repo(module_product.id)
+
+
+@pytest.fixture
+def content_view(module_org, repo):
+    return _content_view(repo['id'], module_org.id)
+
```
Comment (jyejare) [inline tests/foreman/cli/test_docker.py:107]: Use global API based fixtures if not create them there and use it here.

Comment (jyejare) [reply tests/foreman/cli/test_docker.py:107]: Similar setup fixtures you have created in API tests. Creating a global API fixtures can be used in CLI, API and UI tests.

File: tests/foreman/ui/test_contenthost.py [hunk 5]
```diff
@@ -172,7 +172,7 @@ def test_positive_end_to_end(session, repos_collection, vm):
     """Create all entities required for content host, set up host, register it
     as a content host, read content host details, install package and errata.
 
-    :id: f43f2826-47c1-4069-9c9d-2410fd1b622c
+    :id: d2bd7166-7dc2-41ae-bf6b-590f933501d9
```
Comment (jyejare) [inline tests/foreman/ui/test_contenthost.py:175]: Is it a duplicate ID ? Why its changed ?

**Review by jyejare — CHANGES_REQUESTED** [review]
Its a very big Pull Request touches API, CLI just not for DynaConf Conversion bu also has Pytest Conversion.

I suggest you to rename your PR name to `Dynaconf and PyTest Conversion of docker settings/tests`.

**Comment by tpapaioa** [comment]
This PR depends on the pytest conversion PR mentioned in the description.

**Comment by tpapaioa** [comment]
Removed pytest conversion commit, so that only dynaconf conversion commit is included in the PR.

File: tests/foreman/ui/test_contenthost.py (hunk 5 above)
Comment (tpapaioa) [reply tests/foreman/ui/test_contenthost.py]: Removed this accidental change.

File: robottelo/config/base.py [hunk 6]
```diff
@@ -477,10 +477,10 @@ def read(self, reader):
     def validate(self):
         """Validate docker settings."""
         validation_errors = []
-        if not self.docker_image:
-            validation_errors.append('[docker] docker_image option must be provided or enabled.')
-        if not self.external_registry_1:
-            validation_errors.append('[docker] external_registry_1 option must be provided.')
+        if not all(self.__dict__.values()):
```
Comment (mshriver) [inline robottelo/config/base.py:480]: Why the change to making all of the fields required?

Though this may be accurate for the tests, I'd like to hold on enforcing that until the facade layer is removed, and settings file validation is tied to test collection/fixture execution.

Comment (tpapaioa) [reply robottelo/config/base.py]: Ok, removed this change.

**Comment by JacobCallahan** [comment]
Can you include tests results for tests that rely on these settings? Please make sure that the robottelo.properties values are dummy values.

**Comment by tpapaioa** [comment]
With the `[docker]` section of `robottelo.properties` commented out set to dummy values, and the correct settings moved to `conf/docker.yaml`, I ran the following tests that use `external_registry_1` and `private_registry_*`:
```
# pytest -k test_positive_synchronize_private_registry tests/foreman/api/test_repository.py
[...]
collected 229 items / 228 deselected / 1 selected                                                                                                                                                                                            
[...]
tests/foreman/api/test_repository.py .
[...]
=== 1 passed, 228 deselected, 8 warnings in 19.84s ===

# pytest -k test_positive_create_with_long_upstream_name tests/foreman/cli/test_docker.py
[...]
collected 39 items / 38 deselected / 1 selected

tests/foreman/cli/test_docker.py .                                                                                                                                                                                                     [100%]
[...]
=== 1 passed, 38 deselected in 36.02s ===
```

Tests that use the `docker_image` setting are currently broken, and will be fixed in a follow-up PR.

File: conf/docker.yaml.template [hunk 7]
```diff
@@ -0,0 +1,13 @@
+DOCKER:
+  # Name of the libvirt image containing the OS with docker instance set up
+  DOCKER_IMAGE: rhel7-docker-base
```
Comment (jyejare) [inline conf/docker.yaml.template:3]: @tpapaioa @mshriver This image is concerning in new pipeline as the image is being used from static libvirt setup. The setting and code needs to be shifted to suite new pipeline before we execute this in new pipeline.

Either in this Pr or next !

File: conf/docker.yaml.template [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  # External docker registry URL in the format http[s]://<server>[:<port>]
+  # EXTERNAL_REGISTRY_1:
+  # Private docker registry URL in the format http[s]://<server>[:<port>]
+  PRIVATE_REGISTRY_URL: https://registry.hub.docker.com
+  # Private docker registry name
+  PRIVATE_REGISTRY_NAME: username/imagename
```
Comment (jyejare) [inline conf/docker.yaml.template:9]:
```suggestion
  # PRIVATE_REGISTRY_NAME: username/imagename
```

**Review by jyejare — CHANGES_REQUESTED** [review]
Changes requested.

**Comment by jyejare** [comment]
@tpapaioa Do you have separate PR open for pytest conversion which was part of this PR ? OR you planning to raise it later post comments fix ?

File: conf/docker.yaml.template (hunk 7 above)
Comment (tpapaioa) [reply conf/docker.yaml.template:3]: This setting is going to be removed in the pytest conversion PR, which re-factors the tests to use broker.

**Comment by tpapaioa** [comment]
pytest conversion PR:
https://github.com/SatelliteQE/robottelo/pull/8354

File: conf/docker.yaml.template (hunk 8 above)
Comment (tpapaioa) [reply conf/docker.yaml.template]: Done.

**Comment by tpapaioa** [comment]
Rebased on top of latest master, which has docker tests converted to pytest:

```
# pytest -k test_positive_pull_image tests/foreman/cli/test_docker.py
[...]
collected 65 items / 64 deselected / 1 selected

tests/foreman/cli/test_docker.py .
[...]
=== 1 passed, 64 deselected, 2 warnings in 315.07s (0:05:15) ====


# pytest -k test_positive_synchronize_private_registry tests/foreman/api/test_repository.py
[...]
collected 229 items / 228 deselected / 1 selected

tests/foreman/api/test_repository.py .
[...]
=== 1 passed, 228 deselected, 8 warnings in 20.29s ===
```

**Review by jyejare — APPROVED** [review]

**Review by JacobCallahan — APPROVED** [review]
ACK thanks!
